src/import_file.c

```diff
@@ -0,0 +1,66 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   import_file.c                                      :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: gwolf <gwolf@student.42vienna.com>         +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/11/13 15:40:15 by gwolf             #+#    #+#             */
+/*   Updated: 2023/11/20 11:26:33 by gwolf            ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "import_file.h"
+
+t_err	ft_import_file(char *filename, char ***lines)
+{
+	int	fd;
+
+	if (!ft_isvalid_filename(filename))
+		return (ERROR);
+	fd = 0;
+	if (ft_err_open(filename, O_RDONLY, &fd))
+		return (ERROR);
+	if (ft_read_and_split(fd, lines))
+		return (ERROR);
+	if (ft_err_close(fd))
+	{
+		ft_free_char_arr(*lines);
+		return (ERROR);
+	}
+	return (SUCCESS);
+}
+
+t_err	ft_read_and_split(int fd, char ***lines)
+{
+	t_buf	buf;
+
+	if (ft_buf_init(&buf))
+		return (ERROR);
+	if (ft_buf_read_file(&buf, fd))
+	{
+		ft_buf_destroy(&buf);
+		return (ERROR);
+	}
+	errno = 0;
+	*lines = ft_split(buf.str, '\n');
+	if (errno)
+		ft_perror("malloc() failed", errno);
```

Comment:
bit weird but ok. This version is shorter than putting it like 
if (!*lines)
{
ft_perror(blbla);
ft_buf_destroy(&buf);
return (ERROR);
}

Cause of our convention to have each bracket on a new line..

Reply:
I would also need the line `ft_buf_destroy(&buf);` twice since I want to destroy in both cases.
Should I change to longer version? We would have space.

Reply:
Nahh leave it. Just takes a while to grasp what is happening